core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -249,13 +249,15 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
    *
    * '''NOTE''': This method must be invoked by the fiber itself.
    */
-  private def drainQueueWhileRunning(
-    cur0: ZIO.Erased
-  ): ZIO.Erased = {
-    var cur = cur0
+  private def drainQueueWhileRunning(cur0: ZIO.Erased): ZIO.Erased = {
+    if (inbox.unsafeIsEmpty) return cur0
 
+    var cur     = cur0
     var message = inbox.poll()
 
+    // Unfortunately we can't avoid the virtual call to `trace` here
+    if (message ne null) updateLastTrace(cur.trace)
```

Comment:
Seems that this was missing (must have missed it when I worked on #8671) and was causing a test to be flaky

Reply:
👍